spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java

```diff
@@ -103,4 +105,29 @@ default S3Resource upload(String bucketName, String key, InputStream inputStream
 	 * @return downloaded object represented as {@link S3Resource}
 	 */
 	S3Resource download(String bucketName, String key);
+
+	/**
+	 * Creates a signed URL for retrieving an object from S3.
+	 *
+	 * @param bucketName - the bucket name
+	 * @param key - the object key
+	 * @param duration - duration that the URL will work
+	 * @return a {@link URL} representing the signed URL
+	 */
+	URL createSignedGetURL(String bucketName, String key, Duration duration);
+
+	/**
+	 * Creates a signed URL for putting an object into S3.
+	 *
+	 * @param bucketName - the bucket name
+	 * @param key - the object key
+	 * @param duration - duration that the URL will work
+	 * @return a {@link URL} representing the signed URL
+	 */
+	URL createSignedPutURL(String bucketName, String key, Duration duration, @Nullable ObjectMetadata metadata,
```

Comment:
Should we return signedHeaders, SignedPayload, and isBrowserExecutable as well here?